modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/headless_dataset_display/page.jsp

```diff
@@ -0,0 +1,74 @@
+<%--
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+--%>
+
+<%@ include file="/headless_dataset_display/init.jsp" %>
+
+<link href="<%= PortalUtil.getStaticResourceURL(request, PortalUtil.getPathModule() + "/frontend-taglib-clay/dataset_display/styles/main.css") %>" rel="stylesheet" />
+
+<div class="table-root" id="<%= containerId %>">
```

Comment:
We can generate this markup in the `*Tag.java` and then simply use the `getHydratedModuleName` to render the `datsetDisplay` component passing all the information needed